bem server: Move introspection logic of Arch into separate node

scf2k opened the pull request.

arikon committed:
Add util.uniq() generator of callbacks for [].reduce()

arikon committed:
Add ability to customize build rules more flexibly by providing Arch.createCustomNode() method

Commit:
Tests bem-bl submodule is updated to 0.2 (close #264)

- tests bem-bl updated
- tests fixed to inherit i18n.js from js-i to build files with include directives (close #265)

arikon committed:
js+coffee and js+coffee-i tech modules

- Add `js+coffee` and `js+coffee-i` tech modules
- Refactor `js` and `js-i` tech modules
- BundleNode: Use `js+coffee` tech instead of `js` for `BorschikNode` of `js` techs by default
- package: Depends on `coffee-script` and `borschik ~0.0.13`
- tests: Update `bem-bl 0.2` (`i18n.js` tech module)
- tests: Use `js+coffee-i` as `js` tech to build bundles

arikon committed:
Some more tuning and jsdoc

arikon committed:
Fix link

Commit:
README is splat into several files

arikon committed:
Add level config with `md` tech for `docs/`

arikon committed:
Merge pull request #273 from bem/doc-split

`README` is splat into several files in `docs/`

banzalik committed:
Update README.ru.md

опечатка помошью -> помощью

vithar committed:
Merge pull request #275 from banzalik/patch-4

Update README.ru.md

arikon committed:
GitLibraryNode: Fix checkout of the last commit

There was an error checking out last commit of the branch:

- checkout of repos with only one commit didn't work
- checkout of branch was touching files from the last commit, it was affecting build processes, as these files was always updated

arikon committed:
Refactor of require() function in `.bem/make.js` files, it is now fully context aware

arikon committed:
Refactor of require() function in declaration files, it is now fully context aware

arikon committed:
util: Introduce `isFileValid()` and `itTimeValid()` convenient utility functions

The are used in `bem-www` `.bem/make.js`, for example

apsavin committed:
Docs fix: bemhtml.js -> bemhtml

arikon committed:
Merge pull request #276 from apsavin/docs-fix

Docs fix: bemhtml.js -> bemhtml

arikon committed:
Create node_modules/bem -> .. symlink on module install (Close #277)

arikon committed:
Do not make symlink on Travis build run

arikon committed:
Merge branch 'master' into bem-www

Commit:
Part of the default-arch is moved to standalone node (SeedNode) (close #87)
Libraries utilize the same build initialization as project - it's possible to build blocks levels within libraries (close #52)

Commit:
Merge branch 'master' of github.com:bem/bem-tools into arch-node

arikon commented on lib/nodes/seed.js:
```diff
@@ -0,0 +1,176 @@
+    },
+
+    removeRudiments: function(node, keep) {
+        console.dir(keep);
```
`console.log();`

arikon commented on lib/nodes/block.js:
```diff
@@ -27,7 +30,151 @@ registry.decl(BlockNodeName, MagicNode, {
+
+    },
+
+    alterArch: function() {
```
Без копипаста не обойтись?

arikon committed:
Add .editorconfig (Close #278)

arikon committed:
Add .editorconfig: order is necessary

arikon committed:
Refactor Tech: introduce getBuildSuffixesMap()

arikon committed:
Add match*() methods to `simple` level prototype, add tests (Close #282)

arikon committed:
Merge pull request #279 from bem/suffixes-map

Refactor Tech: introduce getBuildSuffixesMap()

arikon committed:
Merge branch 'master' into js+coffee

arikon committed:
Simplify coffee+js.js tech (Ref bem/bem-tools#279)

arikon committed:
Use latest bem-bl with updated i18n.js.js tech to reflect changes in coffee+js.js tech

arikon committed:
Use latest xjst and ometajs to fix tests

- Small cleanup of `js+coffee.js` code
- Use updated `bem-bl`
- Canonize new reference test results

arikon committed:
Merge pull request #268 from bem/js+coffee

js+coffee and js+coffee-i tech modules

arikon committed:
Fix recursion error when build target name contain trailing slash (Close #252)

arikon commented:
@scf2k Please update your branch to make it merge cleanly

arikon committed:
Merge pull request #272 from bem/bem-www

Arch: Add ability to customize build rules more flexibly

arikon committed:
Add `dir` tech module

arikon committed:
Merge pull request #290 from bem/dir-tech

Tech module to create directories

vithar committed:
Simple index page added

vithar committed:
markdown list markup fixed

arikon committed:
Replace own `util.extend()` implementation with `node.extend` module

tadatuta committed:
combine bem commands documentation

tadatuta committed:
Doc: Update index page to fit new doc structure

arikon committed:
Fix jsdoc in util

arikon committed:
Slightly refactor `bem create level` command

- Move validation of level prototype logic to the `COA.Opt.val()` method
- More promisy-based code
- Code cleanup
- Updated reference legacy-tests results

arikon committed:
Use promisy function to write level .js module

arikon committed:
Add ability to `bem create level` command to create level .js modules

arikon committed:
Use coa ~0.3.6

arikon committed:
Fix jsdoc of util.exec(): `options` and `resolveWithOutput` are optional arguments

arikon committed:
Fix jsdoc on util.relative(): mark `dot` argument as optional

arikon committed:
Add util.bemParseKey() function and tests

arikon committed:
Add `.bem` and `node_modules` to `Level.isIgnorablePath()`

arikon committed:
Fix jsdoc of `Tech.getPath()` and `Tech.getPaths()` methods: mark `suffix` and `suffixes` arguments as optional

arikon committed:
A bit of code style fixes in SeedNode

arikon committed:
Remove dead code and minor code cleanup in default-arch

arikon committed:
Remove duplicate code from BundleNode

arikon committed:
Merge branch 'master' into arch-node

Conflicts:
	lib/default-arch.js
	lib/make.js

arikon committed:
Correct bem-bl submodule revision

Commit:
Inspector is improved

The graph is shown collapsed when snapshot is chosen. Click a node to expand/collapse it.

arikon committed:
Fix createCustomNodes() method signature

Commit:
Описал пример поключения локализационных технологий сборки

Commit:
Исправил замечания из #301

Commit:
Добавил английский вариант описания

Commit:
Исправил опечатку в описании

Commit:
Git library checkout fixed to work with commit hashes (close #302)

Parameter branch is added to specify branch name. Use treeish to specify
commit or tag.

tadatuta committed:
Doc: Fix techs link

arikon committed:
Merge branch '0.5.x'

Conflicts:
	package.json

arikon committed:
Fix scripts to use q 0.8.10

arikon committed:
Update npm-shrinkwrap.json

Commit:
Fixed npmPackages check (close #300)

Commit:
Little code clean up

scf2k committed:
Merge pull request #298 from bem/inspector-expand

Arch inspector allows to expand and collapse nodes

arikon committed:
Merge branch '0.5.x'

arikon committed:
Merge pull request #301 from bem/documentation

Описал пример поключения локализационных технологий сборки

arikon committed:
Merge branch '0.5.x'

arikon committed:
Merge branch '0.5.x'

arikon committed:
Use istanbul as a test coverage tool (Close #287)

Commit:
Fixed typo in jsdoc

Commit:
Merge branch 'master' into arch-node

Conflicts:
	lib/nodes/lib.js

scf2k commented:
merged into dev
